File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/PaydayLoans/PaydayLoan.cs
namespace Psi.Data.Models.ClientConfigurationModels.PaydayLoans
{
    public class PaydayLoan : SettingsBaseHelper
    {
        private readonly ISettingsBase _settingsBase;
        private AdvancePay _advancePay;

        public PaydayLoan(ISettingsBase settingsBase) : base(settingsBase)
        {
            _settingsBase = settingsBase;
        }

        public AdvancePay AdvancePay
        {
            get { return _advancePay ?? (_advancePay = new AdvancePay(_settingsBase)); }
            set { _advancePay = value; }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/PaydayLoan.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
import { AdvancePay } from './AdvancePay';
export interface PaydayLoanConfig {
    AdvancePay: AdvancePay;
}

export class PaydayLoan implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'PaydayLoan'
    };


            private _advancePay: AdvancePay;
            get advancePay(): AdvancePay {
                return this._advancePay;
            }
            set advancePay(value: AdvancePay) {
                this._advancePay = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "PaydayLoan.AdvancePay", value: this._advancePay, dataType: 'advancepay', label: "Advance Pay" },
            ];
        }

}